### PR TITLE
Features: meteor/fetch trace

### PR DESCRIPTION
**Pull request by afrokick**
Add tracing for a `meteor/fetch` package

- modernize `http.js`
- rename and export `haveAsyncCallback` function from utils.

### CODE REVIEW

File: lib/hijack/http.js [hunk 1]
```diff
@@ -1,29 +1,83 @@
...
-
-      //if the user supplied an asynCallback, we don't have a response object and it handled asynchronously
-      //we need to track it down to prevent issues like: #3
-      var endOptions = HaveAsyncCallback(arguments) ? { async: true } : { statusCode: response.statusCode };
-      if (eventId) {
-        Kadira.tracer.eventEnd(kadiraInfo.trace, eventId, endOptions);
-      }
+      const response = originalCall.apply(this, arguments);
+
+      const endOptions = { library };
```
Comment (afrokick) [inline lib/hijack/http.js:66]: @zodern Should I add an `async` flag? How it works?

**Comment by afrokick** [comment]
@zodern tests fails for Meteor < 1.7 because

```
* modules@0.16.0 <- modern-browsers 0.1.7 <- fetch 0.1.0 <- local-test:montiapm:agent 2.45.1
```

Any advise how to fix it?

File: lib/hijack/http.js (hunk 1 above)
Comment (zodern) [reply lib/hijack/http.js:66]: Fetch is always async. You can either not have the async field, or have it always be true. Either option should be fine.

**Comment by zodern** [comment]
There might be a better way, but one option is to keep the fetch dependency weak unless the node version is at least 8.11 in `process.versions.node`.

**Comment by afrokick** [comment]
It works! Thanks! But the MongoDB's test failed. Is it flaky test?)

**Comment by zodern** [comment]
> But the MongoDB's test failed. Is it flaky test?)

Yes.

File: lib/hijack/http.js [hunk 2]
```diff
@@ -1,29 +1,82 @@
...
+    }
+  };
+}
+
+if (Package['fetch']) {
+  const library = 'meteor/fetch';
+  const originalCall = Package['fetch'].fetch;
+  const Request = Package['fetch'].Request;
+
+  Package['fetch'].fetch = function (url, opts) {
```
Comment (afrokick) [inline lib/hijack/http.js:47]: As mentioned in meteor/fetch:

https://github.com/meteor/meteor/tree/devel/packages/fetch

> In Node, the [node-fetch](https://www.npmjs.com/package/node-fetch) polyfill is used. Note: unlike the client polyfills, the Node polyfill does not define the fetch function globally. However, any application or package that depends on the Meteor fetch package can refer to fetch as if it was a global function (or import { fetch } from "meteor/fetch").

I have no ideas how to override global fetch

Comment (zodern) [reply lib/hijack/http.js:47]: Meteor takes `Package['fetch'].fetch` and creates a global variable for it. How you are wrapping the fetch package should work, it just will only be wrapped for packages that load after it is wrapped.

Comment (afrokick) [reply lib/hijack/http.js:47]: I just tried this:

`fetch = Package['fetch'] = function ...` and it works. But `fetch` isn't a function ,it is a `module`:

```
I20220303-03:13:45.681(3)? <ref *1> [Function: fetch] {
I20220303-03:13:45.681(3)?   isRedirect: [Function (anonymous)],
I20220303-03:13:45.681(3)?   Promise: [Function: Promise] {
I20220303-03:13:45.681(3)?     Fiber: [Function: Fiber] {
I20220303-03:13:45.681(3)?       yield: [Function (anonymous)],
I20220303-03:13:45.681(3)?       current: [Fiber],
I20220303-03:13:45.681(3)?       poolSize: 120,
I20220303-03:13:45.681(3)?       fibersCreated: 12
I20220303-03:13:45.682(3)?     },
I20220303-03:13:45.682(3)?     awaitAll: [Function (anonymous)],
I20220303-03:13:45.682(3)?     await: [Function (anonymous)],
I20220303-03:13:45.682(3)?     async: [Function (anonymous)],
I20220303-03:13:45.682(3)?     asyncApply: [Function (anonymous)]
I20220303-03:13:45.682(3)?   },
I20220303-03:13:45.682(3)?   default: [Circular *1],
I20220303-03:13:45.682(3)?   Headers: [class Headers],
I20220303-03:13:45.682(3)?   Request: [class Request],
I20220303-03:13:45.682(3)?   Response: [class Response],
I20220303-03:13:45.682(3)?   FetchError: [Function: FetchError]
I20220303-03:13:45.682(3)? }
```

Is it good idea to override it?

**Review by zodern — COMMENTED** [review]
Thanks for working on this! This looks really good.

**Comment by afrokick** [comment]
Sorry for the long delay. Just had a fix for `library` field.

**Comment by afrokick** [comment]
@zodern could you please review?

**Comment by zodern** [comment]
Thanks @afrokick!

**Pull request merged**